protocol/x/clob/keeper/deleveraging.go

```diff
@@ -233,122 +233,137 @@ func (k Keeper) OffsetSubaccountPerpetualPosition(
 	deltaQuantumsRemaining = new(big.Int).Set(deltaQuantumsTotal)
 	fills = make([]types.MatchPerpetualDeleveraging_Fill, 0)
 
-	k.subaccountsKeeper.ForEachSubaccountRandomStart(
-		ctx,
-		func(offsettingSubaccount satypes.Subaccount) (finished bool) {
-			// Iterate at most `MaxDeleveragingSubaccountsToIterate` subaccounts.
-			if numSubaccountsIterated >= k.Flags.MaxDeleveragingSubaccountsToIterate {
-				return true
-			}
-
-			numSubaccountsIterated++
-			offsettingPosition, _ := offsettingSubaccount.GetPerpetualPositionForId(perpetualId)
-			bigOffsettingPositionQuantums := offsettingPosition.GetBigQuantums()
-
-			// Skip subaccounts that do not have a position in the opposite direction as the liquidated subaccount.
-			if deltaQuantumsRemaining.Sign() != bigOffsettingPositionQuantums.Sign() {
-				numSubaccountsWithNoOpenPositionOnOppositeSide++
-				return false
-			}
-
-			// TODO(DEC-1495): Determine max amount to offset per offsetting subaccount.
-			var deltaBaseQuantums *big.Int
-			if deltaQuantumsRemaining.CmpAbs(bigOffsettingPositionQuantums) > 0 {
-				deltaBaseQuantums = new(big.Int).Set(bigOffsettingPositionQuantums)
-			} else {
-				deltaBaseQuantums = new(big.Int).Set(deltaQuantumsRemaining)
-			}
-
-			// Fetch delta quote quantums. Calculated at bankruptcy price for standard
-			// deleveraging and at oracle price for final settlement deleveraging.
-			deltaQuoteQuantums, err := k.getDeleveragingQuoteQuantumsDelta(
-				ctx,
-				perpetualId,
-				liquidatedSubaccountId,
-				deltaBaseQuantums,
-				isFinalSettlement,
+	// Find subaccounts with open positions on the opposite side of the liquidated subaccount.
+	isDeleveragingLong := deltaQuantumsTotal.Sign() == -1
+	subaccountsWithOpenPositions := k.DaemonLiquidationInfo.GetSubaccountsWithOpenPositionsOnSide(
+		perpetualId,
+		!isDeleveragingLong,
+	)
+
+	numSubaccounts := len(subaccountsWithOpenPositions)
+	if numSubaccounts == 0 {
+		liquidatedSubaccount := k.subaccountsKeeper.GetSubaccount(ctx, liquidatedSubaccountId)
+		k.Logger(ctx).Error(
+			"Failed to find subaccounts with open positions on opposite side of liquidated subaccount",
+			"blockHeight", ctx.BlockHeight(),
+			"perpetualId", perpetualId,
+			"deltaQuantumsTotal", deltaQuantumsTotal,
+			"liquidatedSubaccount", liquidatedSubaccount,
+		)
```

Comment:
Suggestion: is it worth noting in the error log that the daemon might have an invalid historical view of the subaccounts with open positions?